api/src/main/java/org/openmrs/module/pihmalawi/sql/MysqlScriptRunner.java

```diff
@@ -0,0 +1,207 @@
+package org.openmrs.module.pihmalawi.sql;
+
+/*
+ * Slightly modified version of ScriptRunner
+ * https://github.com/BenoitDuffez/ScriptRunner
+ */
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.io.PrintWriter;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MysqlScriptRunner {
+
+    private static Log log = LogFactory.getLog(MysqlScriptRunner.class);
+
+    private static final String DEFAULT_DELIMITER = ";";
+    /**
+     * regex to detect delimiter.
+     * ignores spaces, allows delimiter in comment, allows an equals-sign
+     */
+    public static final Pattern delimP = Pattern.compile("^\\s*(--)?\\s*delimiter\\s*=?\\s*([^\\s]+)+\\s*.*$", Pattern.CASE_INSENSITIVE);
+    private final Connection connection;
+
+    private final boolean stopOnError;
+    private final boolean autoCommit;
+
+    private String delimiter = DEFAULT_DELIMITER;
+    private boolean fullLineDelimiter = false;
+
+    public MysqlScriptRunner(Connection connection, boolean stopOnError, boolean autoCommit) {
+        this.connection = connection;
+        this.stopOnError = stopOnError;
+        this.autoCommit = autoCommit;
+    }
+
+    public void setDelimiter(String delimiter, boolean fullLineDelimiter) {
+        this.delimiter = delimiter;
+        this.fullLineDelimiter = fullLineDelimiter;
+    }
+    private String getDelimiter() {
+        return delimiter;
+    }
+    /**
+     * Runs an SQL script (read in using the Reader parameter)
+     *
+     * @param reader - the source of the script
+     */
+    public MysqlResult runScript(Reader reader) throws IOException, SQLException {
+        try {
+            boolean originalAutoCommit = connection.getAutoCommit();
+            try {
+                if (originalAutoCommit != this.autoCommit) {
+                    connection.setAutoCommit(this.autoCommit);
+                }
+                return runScript(connection, reader);
+            } finally {
+                connection.setAutoCommit(originalAutoCommit);
+            }
+        } catch (IOException e) {
+            throw e;
```

Comment:
Can we run this through and fix the formatting to reflect our normal practices by having a line break between a closing-brace and the next line of code throughout?